model/stacktrace.go

```diff
@@ -7,12 +7,52 @@ import (
 
 type Stacktrace []*StacktraceFrame
 
-func (st *Stacktrace) Transform(config *pr.Config, service Service) []common.MapStr {
-	var frames []common.MapStr
+func (st Stacktrace) Transform(config *pr.Config, service Service) []common.MapStr {
+	if config == nil || config.SmapMapper == nil {
+		return st.transform(config)
+	}
+	return st.sourcemapAndTransform(config, service)
+}
 
-	for _, fr := range *st {
-		frame := fr.Transform(config, service)
+func (st Stacktrace) transform(config *pr.Config) []common.MapStr {
+	var frames []common.MapStr
+	for _, fr := range st {
+		frame := fr.Transform(config)
 		frames = append(frames, frame)
 	}
 	return frames
 }
+
+// source map algorithm:
+// apply source mapping frame by frame
+// if no source map could be found, set updated to false and set sourcemap error
+// otherwise use source map library for mapping and update
+// - filename: only if it was found
+// - function:
+//   * should be moved down one stack trace frame,
+//   * the function name of the first frame is set to <anonymous>
+//   * if one frame is not found in the source map, this frame is left out and
+//   the function name from the previous frame is set for the one but next frame
```

Comment:
> previous frame is set for the one but next frame

not sure what was meant here

Reply:
@roncohen I've pushed a commit to hopefully clear this up.  Essentially `fct` stays the same if the mapping can not be applied and is used as the function name for the next frame.